src/lib.rs

```diff
@@ -184,7 +184,23 @@
 //!
 #![deny(missing_docs)]
 
-use std::collections::hash_map::RandomState;
+#[cfg(feature = "parking_lot")]
+extern crate parking_lot;
+
+#[cfg(feature = "hashbrown")]
+extern crate hashbrown;
+
+#[cfg(feature = "smallvec")]
+extern crate smallvec;
+
+/// Re-export default hash builder
+#[cfg(feature = "hashbrown")]
+pub type DefaultHashBuilder = hashbrown::hash_map::DefaultHashBuilder;
+
+/// Re-export default hash builder
+#[cfg(not(feature = "hashbrown"))]
+pub type DefaultHashBuilder = std::collections::hash_map::RandomState;
```

Comment:
Hmm, I don't think this is okay, because Rust features [must be additive](https://github.com/rust-lang/cargo/issues/4323). In this case, imagine two crates, `a` and `b`, that both depend on `evmap`, but one does without the `hashbrown` feature, and one does with. Cargo will merge them into a single dependence on `evmap` with the feature enabled, at which point `a` won't compile since the types potentially won't match (if they explicitly name `Options<(), std::collections::hash_map::RandomState>` for example. I'm not entirely sure how to work around that though... Perhaps we just make `hashbrown` a non-optional dependency?

Reply:
Hmm. Nothing really changes if they manually specify a hash builder, like with `Options::with_builder`.

Although not entirely additive, we could make it clear in the docs/reader that swapping out the standard hashmap for hashbrown does also swap between the default hash builder for both. 

Then again, with hashbrown being considered as a real replacement in the standard library, making it non-optional is also reasonable, but would be a breaking change. If the feature is left disabled right now the new type is still just `RandomState`.

Reply:
No, the issue is if they use the default, but have written down the explicit type somewhere. Now, that said, doing so is probably rare if they're using the default random state, so maybe it's fine. As a somewhat long-shot, I'll ping @kennytm here to see if they have any thoughts, since they know far more about this than I do. I think I'm leaning towards saying that the change is fine as-is, though a comment in the docs is probably a worthwhile addition.

Reply:
Yeah, I'm also leaning towards 2, though would also be okay with 1. One option would be to do 1 under the assumption that the `std` `HashMap` will change to `hashbrown` in the future. Have you had a chance to benchmark the `hashbrown` swap in isolation to see that it actually speeds things up markedly?

Reply:
After some thought, 2 would be the least breaking change, but we can re-export `hashbrown`'s FxHash implementation to make it obvious it exists. So I'm going to go with that.